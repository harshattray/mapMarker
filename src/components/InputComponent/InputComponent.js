/**
 * @Author: harsha
 * @Date:   2019-04-17T23:57:06+05:30
 * @Last modified by:   harsha
 * @Last modified time: 2019-04-24T23:59:51+05:30
 */

import React, { Fragment, Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Button } from "semantic-ui-react";
import { reduxForm, Field, reset } from "redux-form";
import { RenderInputFields } from "../renderInputFieldComponent/renderInputFieldComponent";
import { submitFormData } from "../../actions/SearchActions";
import { validate } from "../../helpers/validate";

/**
 * [InputComponent handles both location search and edit location]
 * @extends Component
 */

class InputComponent extends Component {
  /**
   * [handleLocationSubmit submit function triggers submitFormData based on params passed to it]
   * @param  {[type]} data [description]
   * @return {[type]}      [description]
   */
  handleLocationSubmit = data => {
    if (this.props.point) {
      this.props.submitFormData(data, this.props.point, this.props.index);
    } else {
      this.props.submitFormData(data);
    }
  };
  render() {
    const { handleSubmit, point, invalid, submitting } = this.props;
    return (
      <form
        name="locationSearchForm"
        onSubmit={handleSubmit(this.handleLocationSubmit)}
        noValidate
        className="form-section"
      >
        <Field
          name="location"
          component={RenderInputFields}
          placeholder={
            point ? "Enter Location to be updated" : "Enter Location here"
          }
          type="text"
          required
          label={point ? "Edit Location" : "Enter Location"}
          className="column"
        />
        <Button className="search" disabled={invalid} loading={submitting}>
          Search
        </Button>
      </form>
    );
  }
}
function mapStateToProps({ searchData, form }) {
  return {
    searchResults: searchData.getResultsStack
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ submitFormData }, dispatch);
}

const afterSubmitdata = (result, dispatch) =>
  dispatch(reset("locationSearchForm"));

InputComponent = reduxForm({
  validate,
  form: "locationSearchForm",
  destroyOnUnmount: false,
  onSubmitSuccess: afterSubmitdata
})(InputComponent);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(InputComponent);
