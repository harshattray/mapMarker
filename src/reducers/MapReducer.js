/**
 * @Author: harsha
 * @Date:   2019-04-16T01:41:56+05:30
 * @Last modified by:   harsha
 * @Last modified time: 2019-04-24T02:59:24+05:30
 */

import { SET_MAP_OBJECT } from "../actions/types";

/**
 * [Map reducer]
 * @param  {Array}  [state=[]] [description]
 * @param  {Object} action     [description]
 * @return {[type]}            [description]
 */

export default (state = [], action) => {
  switch (action.type) {
    case SET_MAP_OBJECT:
      return {
        ...state,
        initialMapData: action.payload
      };
    default:
      return state;
  }
};
