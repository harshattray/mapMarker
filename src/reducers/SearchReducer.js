/**
 * @Author: harsha
 * @Date:   2019-04-18T00:48:33+05:30
 * @Last modified by:   harsha
 * @Last modified time: 2019-04-25T02:50:40+05:30
 */
import {
  GET_RESULTS_STACK,
  EDIT_RESULTS_STACK,
  CANCEL_EDITING,
  SET_MAP_OBJECT,
  IS_EDITING,
  DELETE_MARKER
} from "../actions/types";
import {
  mapCoOrdsStack,
  editMarkersArray,
  editPositionCoords,
  deleteMarkersArray,
  markerArrayBuilder,
  deleteCoords
} from "../helpers/MapObjBuilder";

/**
 * [initial_state State object with initial values]
 * @type {Object}
 */

export const initial_state = {
  markersArray: [],
  coOrdsArray: [],
  isEditing: false
};

/**
 * [Search reducer]
 * @param  {Object} [state=initial_state] [initial state object]
 * @param  {Object} action                [action object with types and payload]
 * @return {Object}                       [description]
 */

export default (state = initial_state, action) => {
  switch (action.type) {
    case SET_MAP_OBJECT:
      return {
        ...state,
        initialMapData: action.payload
      };
    case GET_RESULTS_STACK:
      return {
        ...state,
        getResultsStack: action.payload,
        markersArray: state.markersArray.concat(action.marker),
        coOrdsArray: [...state.coOrdsArray, action.payload]
      };
    case EDIT_RESULTS_STACK:
      return {
        ...state,
        getResultsStack: action.payload,
        editIndexValue: action.editingIndex,
        markersArray: editMarkersArray(
          action.editingIndex,
          action.payload.geometry,
          state.markersArray,
          action.payload,
          action.marker
        ),
        coOrdsArray: editPositionCoords(
          state.coOrdsArray,
          action.editingIndex,
          action.payload
        )
      };
    case IS_EDITING:
      return {
        ...state,
        isEditing: action.isEditing,
        pointValues: action.pointValues,
        indexValues: action.indexValues
      };
    case CANCEL_EDITING:
      return {
        ...state,
        isEditing: action.isEditing
      };
    case DELETE_MARKER:
      return {
        ...state,
        deleteIndexValue: action.deletingIndex,
        markersArray: deleteMarkersArray(
          action.deletingIndex,
          state.markersArray
        ),
        coOrdsArray: deleteCoords(action.deletingIndex, state.coOrdsArray)
      };

    default:
      return state;
  }
};
