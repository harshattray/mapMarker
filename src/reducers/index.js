/**
 * @Author: harsha
 * @Date:   2019-04-16T00:57:16+05:30
 * @Last modified by:   harsha
 * @Last modified time: 2019-04-22T05:39:36+05:30
 */

import { combineReducers } from "redux";
import { reducer as formReducer } from "redux-form";
import SearchReducer from "./SearchReducer";

export default combineReducers({
  form: formReducer,
  searchData: SearchReducer
});
