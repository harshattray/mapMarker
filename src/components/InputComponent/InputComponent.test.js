/**
 * @Author: harsha
 * @Date:   2019-04-25T01:05:55+05:30
 * @Last modified by:   harsha
 * @Last modified time: 2019-04-25T01:59:13+05:30
 */

import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import InputComponent from "./InputComponent";
import { createStore } from "redux";
import { Provider } from "react-redux";
import reducers from "../../reducers";

configure({ adapter: new Adapter() });

describe("MapComponent", () => {
  test("renders", () => {
    const wrapper = shallow(
      <Provider store={createStore(reducers, {})}>
        <InputComponent />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
