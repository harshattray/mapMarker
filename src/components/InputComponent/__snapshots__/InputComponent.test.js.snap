// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapComponent renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <UNDEFINED />
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <UNDEFINED />,
      "value": Object {
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "subscription": Subscription {
          "handleChangeWrapper": [Function],
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "destroyOnUnmount": false,
            "enableReinitialize": false,
            "forceUnregisterOnUnmount": false,
            "form": "locationSearchForm",
            "getFormState": [Function],
            "keepDirtyOnReinitialize": false,
            "onSubmitSuccess": [Function],
            "persistentSubmitErrors": false,
            "pure": true,
            "shouldAsyncValidate": [Function],
            "shouldError": [Function],
            "shouldValidate": [Function],
            "shouldWarn": [Function],
            "submitAsSideEffect": false,
            "touchOnBlur": true,
            "touchOnChange": false,
            "updateUnregisteredFields": false,
            "validate": [Function],
          },
          "displayName": "ReduxForm",
          "render": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
        "compare": null,
        "displayName": "Connect(ReduxForm)",
        "type": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <UNDEFINED />,
        "value": Object {
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "subscription": Subscription {
            "handleChangeWrapper": [Function],
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "unsubscribe": [Function],
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "destroyOnUnmount": false,
              "enableReinitialize": false,
              "forceUnregisterOnUnmount": false,
              "form": "locationSearchForm",
              "getFormState": [Function],
              "keepDirtyOnReinitialize": false,
              "onSubmitSuccess": [Function],
              "persistentSubmitErrors": false,
              "pure": true,
              "shouldAsyncValidate": [Function],
              "shouldError": [Function],
              "shouldValidate": [Function],
              "shouldWarn": [Function],
              "submitAsSideEffect": false,
              "touchOnBlur": true,
              "touchOnChange": false,
              "updateUnregisteredFields": false,
              "validate": [Function],
            },
            "displayName": "ReduxForm",
            "render": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
          "compare": null,
          "displayName": "Connect(ReduxForm)",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
